### PR TITLE
[CDAP-20253] improve lcm runtimeargs ux

### PR DESCRIPTION
**Pull request by sumengwang**
# [CDAP-20253] improve lcm runtimeargs ux

## Description
* improve transformation pushdown runtimeargs experience, saving from configs tab will delete outdated settings
* bind pushdown configurations on page load, so that settings will not always be empty when refreshing the page
* some other small bug fixes, added more comments

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [x] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-20253](https://cdap.atlassian.net/browse/CDAP-20253)

[CDAP-20253]: https://cdap.atlassian.net/browse/CDAP-20253?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDAP-20253]: https://cdap.atlassian.net/browse/CDAP-20253?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/cdap/components/PipelineConfigurations/ConfigurationsContent/PushdownTabContent.tsx [hunk 1]
```diff
@@ -34,11 +39,29 @@ const getPushdownEnabledValue = (state) => {
     : state.pushdownEnabled;
 };
 
+const getTransformationPushdownValue = (state) => {
+  const transformationPushdownKeyValuePair = state.runtimeArgs.pairs.filter((pair) =>
+    pair.key.startsWith(GENERATED_RUNTIMEARGS.PIPELINE_TRANSFORMATION_PUSHDOWN_PREFIX)
+  );
+  if (state.transformationPushdown) {
```
Comment (mrahanjam) [inline app/cdap/components/PipelineConfigurations/ConfigurationsContent/PushdownTabContent.tsx:43]: nit: move this to the top of the function for an earlier exit if it checks.

File: app/cdap/components/PipelineConfigurations/ConfigurationsContent/ResourcesTabContent/DriverResources.js [hunk 2]
```diff
@@ -59,10 +59,12 @@ const mapDispatchToProps = (dispatch) => {
         type: PipelineConfigurationsActions.SET_DRIVER_VIRTUAL_CORES,
         payload: { virtualCores: e.target.value },
       });
-      dispatch({
-        type: PipelineConfigurationsActions.SET_RUNTIME_ARGS,
-        payload: { runtimeArgs: { pairs: newRunTimePairs } },
-      });
+      if (lifecycleManagementEditEnabled) {
```
Comment (mrahanjam) [inline app/cdap/components/PipelineConfigurations/ConfigurationsContent/ResourcesTabContent/DriverResources.js:57]: A little context would help here. Looks like we're setting runtime args only when `lifecycleManagementEditEnabled` is true? what happens to the usual scenario without lifecycle management?

**Review by mrahanjam — APPROVED** [review]
LGTM overall, left a nit and a question

File: app/cdap/components/PipelineConfigurations/Store/ActionCreator.js [hunk 3]
```diff
@@ -152,20 +149,20 @@ const updatePreferences = (
     const customSparkConfigKeyValuePairs = runtimeArgs.pairs.filter((pair) =>
       pair.key.startsWith(GENERATED_RUNTIMEARGS.CUSTOM_SPARK_KEY_PREFIX)
     );
-    let pairs = cloneDeep(customSparkConfigKeyValuePairs);
-
-    // engine config overwritting
-    if (!overwriteEngineConfig) {
-      // save from runtime args dropdown
-      pairs.forEach((pair) => {
+    let customSparkConfigPairs = cloneDeep(customSparkConfigKeyValuePairs);
```
Comment (seanfeld) [inline app/cdap/components/PipelineConfigurations/Store/ActionCreator.js]: looks like you're just adding to this array right? then it can be const

**Review by seanfeld — APPROVED** [review]
just a nit

**Review by radhikav1 — APPROVED** [review]

**Pull request merged**